### PR TITLE
[Translations] [Indonesian]: /Id/advanced/deployment.md

### PR DESCRIPTION
**Pull request by CikiMomogi**
Ready for review.

### CODE REVIEW

File: site/docs/id/advanced/deployment.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+1. [Pasang `error handler` menggunakan `bot.catch` (long polling) atau pada web framework-mu (webhooks).](../guide/errors.md)
+2. Gunakan `await` di semua `promise` dan install **linting**, yang mewajibkan penerapan aturan tersebut, dengan begitu kamu tidak akan lupa untuk menggunakan dua syntax ini.
+
+## Pengiriman Pesan
+
+1. Kirim file mengunakan `path` atau `buffer`, bukan dengan `stream`, atau setidaknya pastikan Kamu [tahu jebakannya](./transformers.md#use-cases-of-transformer-functions).
+2. Gunakan `bot.on("callback_query:data")` sebagai penanganan _fallback_ untuk [menanggapi semua `callback query`](../plugins/keyboard.md#responding-to-clicks).
+3. Gunakan [plugin `transformer-throttler`](../plugins/transformer-throttler.md) agar terhindari dari rate limit.
+4. **Opsional.** Gunakan [plugin `auto-retry`](../plugins/auto-retry.md) untuk menangani error `flood wait` secara otomatis.
```
Comment (CikiMomogi) [inline site/docs/id/advanced/deployment.md:22]: - [ ] Fix these anchor links

File: site/docs/id/advanced/deployment.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Penskalaan
+
+Ini tergantung dari jenis deployment kamu.
+
+### Long Polling
+
+1. [Gunakan grammY runner.](../plugins/runner.md),
+2. [Gunakan `sequentialize` dengan _session key_ dari _resolver function_ yang sama sebagai _session middleware_ kamu](./scaling.md#concurrency-itu-sulit),
```
Comment (CikiMomogi) [inline site/docs/id/advanced/deployment.md:31]: - [ ] Fix anchor link

File: site/docs/id/advanced/deployment.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+3. Periksa konfigurasi `run` ([Referensi API](https://doc.deno.land/https://deno.land/x/grammy_runner/mod.ts/~/run)) dan pastikan sesuai dengan kebutuhanmu, atau bahkan pertimbangkan untuk membuat runner-mu sendiri dari source dan sink.
+   Hal utama yang perlu dipertimbangkan adalah beban maksimum yang ingin diterapkan ke servermu, misal: berapa banyak update yang dapat diproses secara bersamaan,
+4. Implementasikan [graceful shutdown](./reliability.md#graceful-shutdown) ketika hendak menghentikan bot (misalnya untuk beralih ke versi baru).
```
Comment (CikiMomogi) [inline site/docs/id/advanced/deployment.md:34]: - [ ] Fix anchor link

File: site/docs/id/advanced/deployment.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+### Webhooks
+
+1. Pastikan tidak melakukan operasi yang berjalan lama di middleware, seperti pengiriman file dalam jumlah besar.
+   [Hal ini akan mengakibatkan error timeout](../guide/deployment-types.md#ending-webhook-requests-in-time) pada webhooks, dan pemrosesan update berulang kali dikarenakan Telegram akan terus mengirim kembali update yang tidak direspon.
```
Comment (CikiMomogi) [inline site/docs/id/advanced/deployment.md:39]: - [ ] Fix anchor link

File: site/docs/id/advanced/deployment.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+   Untuk menghindari hal tersebut, sebaiknya gunakan _task queuing system_,
+2. Buat dirimu terbiasa dengan konfigurasi `webhookCallback` ([Referensi API](https://doc.deno.land/https://deno.land/x/grammy/mod.ts/~/webhookCallback)),
+3. Jika mengatur opsi `getSessionKey` untuk session, [Gunakan `sequentialize` dengan _session key_ dari _resolver function_ yang sama sebagai _session middleware_ kamu.](./scaling.md#concurrency-itu-sulit),
```
Comment (CikiMomogi) [inline site/docs/id/advanced/deployment.md:42]: - [ ] Fix anchor link

File: site/docs/id/advanced/deployment.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+4. Jika menjalankan bot di platform serverless atau autoscaling, [atur informasi bot](https://doc.deno.land/https://deno.land/x/grammy/mod.ts/~/BotConfig) untuk mencegah panggilan `getMe` yang berlebihan,
+5. Pertimbangkan untuk menggunakan [webhook reply](../guide/deployment-types.md#webhook-reply).
```
Comment (CikiMomogi) [inline site/docs/id/advanced/deployment.md:44]: - [ ] Fix anchor link

File: site/docs/id/advanced/deployment.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+## Session
+
+1. Pertimbangkan menggunakan `lazySessions` seperti yang sudah dijelaskan [di sini](../plugins/session.md#lazy-sessions).
```
Comment (CikiMomogi) [inline site/docs/id/advanced/deployment.md:48]: - [ ] Fix anchor link

File: site/docs/id/advanced/deployment.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+3. Gunakan [plugin `transformer-throttler`](../plugins/transformer-throttler.md) agar terhindari dari rate limit.
+4. **Opsional.** Gunakan [plugin `auto-retry`](../plugins/auto-retry.md) untuk menangani error `flood wait` secara otomatis.
+
+## Penskalaan
+
+Ini tergantung dari jenis deployment kamu.
+
+### Long Polling
+
+1. [Gunakan grammY runner.](../plugins/runner.md).
```
Comment (fadzikri) [inline site/docs/id/advanced/deployment.md:30]:
```suggestion
1. [Gunakan grammY runner](../plugins/runner.md).
```

Comment (fadzikri) [reply site/docs/id/advanced/deployment.md:30]: Titiknya ikutan jadi link dan itupun jadi ada 2

**Review by fadzikri — CHANGES_REQUESTED** [review]
Itu saja sih.

**Review by fadzikri — APPROVED** [review]

File: site/docs/id/advanced/deployment.md (hunk 8 above)
Comment (CikiMomogi) [reply site/docs/id/advanced/deployment.md:30]: Typo. Good catch!

**Pull request merged**